extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/ProgrammaticApiTest.java

```diff
@@ -0,0 +1,126 @@
+package io.quarkus.cache.test.runtime;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.cache.Cache;
+import io.quarkus.cache.CacheInvalidate;
+import io.quarkus.cache.CacheInvalidateAll;
+import io.quarkus.cache.CacheManager;
+import io.quarkus.cache.CacheResult;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ProgrammaticApiTest {
+
+    private static final String CACHE_NAME = "test-cache";
+    private static final Object KEY_1 = new Object();
+    private static final Object KEY_2 = new Object();
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest().setArchiveProducer(
+            () -> ShrinkWrap.create(JavaArchive.class).addClass(CachedService.class));
+
+    @Inject
+    CachedService cachedService;
+
+    @Inject
+    CacheManager cacheManager;
```

Comment:
The main usage for a user is more likely 

`@Inject Cache cache;`
//should we have a @Cache or @CacheName annotation to qualify?

Reply:
I love the idea! Injections are now possible. More details below in a few minutes.

Reply:
Or just `@Named` ? (from javax.inject)

Reply:
Plus 1 for `@CacheName`. `@Named` has two consequences: (1) the bean is automatically available via EL, in templates, etc. (2) `@Named` is a qualifier but it does not count when the CDI container decides to add the `@Default` qualifier or not. In other words, a bean annotated with `@Named` can be injected into an injection point without a qualifier (where the `@Default` is considered). Finally, the `@Named` at injection points is considered a bad practice (see https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#named_at_injection_point).

Reply:
`@CacheName` is already available (but not yet documented) on master so I'm resolving this comment.